### PR TITLE
Add support for multiple views on Shields

### PR DESCRIPTION
**Pull request by cezaraugusto**
close https://github.com/brave/brave-browser/issues/1196
close https://github.com/brave/brave-browser/issues/4953

shields-views

see https://brave-ui-4200l5i5w.now.sh/?path=/story/feature-components-shields--panel

### Test Plan 
See https://github.com/brave/brave-browser/issues/1196 for design spec

1. Initial state for Shields should be the "simple view" (no option for controlling resources blocked)
2. First time switching to "advanced view" should show a warning. This warning should show only. one time, i.e. toggling views again should not show it.
3. Should be able to toggle simple/advanced views via link in Shields footer.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by cezaraugusto** [comment]
@karenkliu updated

File: components/test/brave_extension/background/reducers/cosmeticFilterReducer_test.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ describe('cosmeticFilterReducer', () => {
       shieldsPanelReducer(initialState.shieldsPanel, {
         type: windowTypes.WINDOW_REMOVED,
         windowId
-      })
+      })
```
Comment (petemill) [inline components/test/brave_extension/background/reducers/cosmeticFilterReducer_test.ts]: nit: whitespace

File: components/test/brave_extension/state/shieldsPanelState_test.ts [hunk 2]
```diff
@@ -196,6 +226,9 @@ describe('shieldsPanelState test', () => {
     it('updates the currentWindowId', () => {
       expect(shieldsPanelState.updateFocusedWindow(state, 2)).toEqual({
         currentWindowId: 2,
+        persistentData: {
```
Comment (petemill) [inline components/test/brave_extension/state/shieldsPanelState_test.ts]: nit: this all seems quite repetitive! Perhaps a function `getDefaultState` would remove the need to copy and keep in sync amongst the tests, next time we need to update this :-)

File: components/test/brave_extension/background/reducers/cosmeticFilterReducer_test.ts (hunk 1 above)
Comment (cezaraugusto) [reply components/test/brave_extension/background/reducers/cosmeticFilterReducer_test.ts]: updated

File: components/test/brave_extension/state/shieldsPanelState_test.ts (hunk 2 above)
Comment (cezaraugusto) [reply components/test/brave_extension/state/shieldsPanelState_test.ts]: are you ok if we address this in a follow-up? I've been looking to re-work some shields tests for a while, can add this as well

Comment (petemill) [reply components/test/brave_extension/state/shieldsPanelState_test.ts]: Absolutely, it was just a comment

File: components/brave_extension/extension/brave_extension/_locales/en_US/messages.json [hunk 3]
```diff
@@ -158,5 +158,33 @@
   "resetAllFilterSettings": {
     "message": "Clear CSS rules for all sites",
     "description": "Message for context menu that resets all cosmetic filters"
+  },
+  "advancedView": {
+    "message": "Advanced View",
+    "description": "Message for the advannced view option"
```
Comment (bsclifton) [inline components/brave_extension/extension/brave_extension/_locales/en_US/messages.json]: s/advannced/advanced

File: components/brave_extension/extension/brave_extension/_locales/en_US/messages.json [hunk 4]
```diff
@@ -158,5 +158,33 @@
...
+  },
+  "simpleView": {
+    "message": "Simple View",
+    "description": "Message for the simple view option"
+  },
+  "shieldsExplanation": {
+    "message": "Sites often include cookies and scripts which try to identify you and your deviec (often embedded into ads). They want to work out who you are and follow you accross the web — tracking what you do on every site. Brave blocks these things so that you can browse without being followed around.",
```
Comment (bsclifton) [inline components/brave_extension/extension/brave_extension/_locales/en_US/messages.json]: s/deviec/device

(might be good to drop all the text in an editor w/ spellcheck)

**Pull request merged**